components/base/kinematicbase/ptgKinematics.go

```diff
@@ -154,6 +176,9 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 			return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
 		}
 		utils.SelectContextOrWait(ctx, timestep)
+		ptgk.inputLock.Lock() // In the case where there's actual contention here, this could cause timing issues; how to solve?
+		ptgk.currentInput = []referenceframe.Input{inputs[0], inputs[1], {trajNode.Dist}}
```

Comment:
[q] if this persisted, and was not overwritten once we returned, couldn't `ptgk.currentInputs()` be used to give us the actual element of the plan we had just executed?
couldn't this be used instead of the `currentNode` value in `errorState`